src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -465,6 +466,17 @@ TaskActivity INameVersionObjectManager<TaskActivity>.GetObject(string name, stri
             return new TaskActivityShim(this, info.Executor, this.hostLifetimeService, name);
         }
 
+        private Task ActivityMiddleware(DispatchMiddlewareContext dispatchContext, Func<Task> next)

```

Comment:
Nitpick:
I find the `Middleware`-suffixed method names to be a bit mysterious, in part because they're not given much documentation. I would really appreciate a header comment here to that can clarify
(1) how to think of the dispatchContext 
(2) how the function works, especially around its `return next()`  line (line 477)

While I realize this is not strictly related to the feature, the header comment would help illuminate other features that re-use the "AddMiddleware" pattern.

Reply:
Good idea. I'm probably the only one who understands this stuff so more documentation would be helpful. :)